### DISCUSSION TITLE
Usability for python generation?

### DISCUSSION THREAD

**Discussion by ghost** [LostRuins/koboldcpp#232] [Q&A]
My only use case is for generation of python code.

Ooba is getting bloated.. and formatting of outputted code format seems broken ( again ). Tried koboldccp again ( been a while .. now i remember why i switched ) and it seems that while it works fine for regular chat, any code output is just a big blob, no indentation, etc.  Lot of effort to clean up.

Are there any formatting options i'm missing so that code that is generated is usable as is?

And, the other reason i switched, how does one 'continue' from where it left off when your max tokens runs out? Is that possible now?

**Comment by LostRuins** [comment]
There is some markdown support which you can enable in the settings for instruct mode, which should help by adding code blocks and limited syntax styling.

**Comment by LostRuins** [comment]
As for context size management that should be mostly automatic

**Comment by ghost** [comment]
Thank you i will look for that.

EDIT: adding markdown for instruct did help format, so thank you, but ran into the same problem as before with continuation when generation tokens run out.  It just starts over from the beginning if i try to get it to continue via a prompt. That was one of the great things about ooba when it came out, a continue button.. that works.
